lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -976,15 +976,21 @@ def _system_set(config_item: str, value: str) -> None:
         raise SnapError("Failed setting system config '{}' to '{}'".format(config_item, value))
 
 
-def hold_refresh(days: int = 90) -> bool:
+def hold_refresh(days: int = 90, forever: bool = False) -> bool:
     """Set the system-wide snap refresh hold.
 
     Args:
         days: number of days to hold system refreshes for. Maximum 90. Set to zero to remove hold.
+        forever: if set  to True, will set a hold forever.
     """
+    if not isinstance(forever, bool):
+        raise ValueError("forever must be a boolean")
```

Comment:
Personally I'd don't think the runtime type check here will add anything. But if you want to keep it, this should be a TypeError, not a ValueError. And we should probably say "bool" rather than "boolean" to match the Python type name.

```suggestion
        raise TypeError("forever must be a bool")
```